code/modules/reagents/newchem/food.dm

```diff
@@ -133,17 +133,19 @@ datum/reagent/honey/reaction_turf(var/turf/T, var/volume)
 	reagent_state = LIQUID
 	color = "#2E2418"
 
-datum/reagent/chocolate/on_mob_life(var/mob/living/M as mob)
+/datum/reagent/chocolate/on_mob_life(var/mob/living/M as mob)
 	if(!M) M = holder.my_atom
 	M.reagents.add_reagent("sugar", 0.8)
 	..()
 	return
 
-datum/reagent/chocolate/reaction_turf(var/turf/T, var/volume)
+/datum/reagent/chocolate/reaction_turf(var/turf/T, var/volume)
 	src = null
-	if(volume >= 5)
-		new /obj/item/weapon/reagent_containers/food/snacks/reagentchocolatebar(T)
-		return
+	var/volume_left = volume
+	while(volume_left >= 5)
```

Comment:
Multiplying the reagent with smoke was not actually something I had considered. I had only been thinking it was silly that 50 units would result in the same amount of cocoa as 5 units. I had also decreased the amount of reagents in the item because you could convert 5 units into 10 (plus an additional 10 pure sugar that came out of nowhere), so I figured it was fair to balance it out so that you only lost the excess amount that didn't result in another pile.

All that said, this honestly isn't a huge deal and can easily be changed back if there is an agreement on it.